### DISCUSSION TITLE
ESPNOW File with metawrite

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#842] [Q&A] [answered]
how to use metawrite if we are sending file to espnow ? I want to send some other information with file .
.
.

ESPNowStream now;
File audioFile;
StreamCopy copier(now, audioFile);

**Comment by pschatzmann** [comment, marked as answer]
Just look at the class documentation, there is a method to write metadata and a callback method to receive

**Reply by aqildad-create** [reply]
I am in binarycontainer  were their is function of write meta  but  I am confuse how can  use it with file
![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/92d463cf-163f-48a5-8392-a83557de3dbe)

**Comment by pschatzmann** [comment]
What you mean with "use it in file" ? You can put any information there as you want to make available to the receiver: You could put it into a comma delimited string, or a C structure...
It is up to use what and how you want to represent the data. Just cast the data to a const uint_t* and provide the length of the data that you provided

**Reply by aqildad-create** [reply]
use it in file  means i am sending 2 type of data to espnow 
1: encoded data 
2: data from file (wave and mp3 ) ---this is with file 

and i need to add small amount of other data with it and at receiver end  i will use t . 


and  i agree with you and  want to implement this but which class documentation you are referring to ?  

 i  cant able to find the class and further cant find any example related to this .

**Reply by pschatzmann** [reply]
I don't understand: just above you pasted the class documentation from the BinaryContainerEncoder!
What is so difficult just to call e.g to call writeMeta("wav",3) ?

Feel free to extend the examples and provide a pull request

**Reply by aqildad-create** [reply]
BinaryContainerEncoder rr;
EncodedAudioStream encoder( &now, new SBCEncoder() );
.
uint8_t myBufferbuffer[300];
rr.writeMeta(myBufferbuffer, 10) ;

now how to pas it to espnow ?

**Comment by pschatzmann** [comment]
It I remember right I have sent you this example already twice or three times: https://www.pschatzmann.ch/home/2023/04/28/arduino-audio-tools-codecs-and-containers/

**Reply by aqildad-create** [reply]
yes i agree , you have send this example  to me  and also i studied  this but i cant succeed to write meta in this example . 

If you guide and provide  extended version of this example i will really help full .

also i want to use this extended functionality but  have no success . please help 
![image](https://github.com/pschatzmann/arduino-audio-tools/assets/65456998/e11b52dd-9728-449b-8f1a-0a0c853f86c2)

**Reply by aqildad-create** [reply]
I try multiple attempts  but still no success . help and your attention is needed .

**Reply by aqildad-create** [reply]
****You can write you own additional (meta) data in the encoder with writeMeta(const uint8_t *data, size_t len) and retrieve it in the decoder with a callback method.**** 

From above i personally conclude write meta to encoder so  i  try below but no success still 

 encoder.writeMeta(myBufferbuffer, 10) ;
